pandas/core/generic.py

```diff
@@ -6840,6 +6839,9 @@ def interpolate(
         elif axis == 1:
             _maybe_transposed_self = self.T
             ax = 1
+        else:
+            msg = f"axis must be 0 or 1, {axis} was provided"
+            raise ValueError(msg)
```

Comment:
I think should raise here because otherwise `_maybe_transposed_self` isn't defined

Reply:
this should not be reachable; _get_axis_number above should raise

Reply:
Ah, you're right